src/main/drivers/dshot.c

```diff
@@ -94,10 +94,10 @@ uint16_t dshotConvertToExternal(float motorValue)
         } else if (motorValue <= DSHOT_3D_FORWARD_MIN_THROTTLE - 1) {
             externalValue = scaleRange(motorValue, DSHOT_MIN_THROTTLE, DSHOT_3D_FORWARD_MIN_THROTTLE - 1, PWM_RANGE_MIDDLE - 1, PWM_RANGE_MIN);
         } else {
-            externalValue = scaleRange(motorValue, DSHOT_3D_FORWARD_MIN_THROTTLE, DSHOT_MAX_THROTTLE, PWM_RANGE_MIDDLE + 1, PWM_RANGE_MAX);
+            externalValue = scaleRange(motorValue + 1, DSHOT_3D_FORWARD_MIN_THROTTLE, DSHOT_MAX_THROTTLE, PWM_RANGE_MIDDLE + 1, PWM_RANGE_MAX);
```

Comment:
Do you have an explanation for why the `+ 1` here is the right thing to do?

If not then I think this should really be using `lrintf(motorValue)`, and that the discrepancy is caused by a truncation being used for `float motorValue`.

Reply:
PWM is used as index in mapping between motorValue and externalValue. Allthough motorValue is a float I always see only integer values?
On line 97 `+1` adjusts mapping to the `+1` in `PWM_RANGE_MIDDLE + 1`. 
For line 100 `+1` adjusts mapping to the `+1` in `PWM_RANGE_MIN + 1`.

Reply:
🤔 Hmm, is this the culprit? Other way around. We can just remove the -1 from `dshotConvertFromExternal()`.
 That way we don;t have to account for the subtraction in `dshotConvertToExternal()` !

Reply:
Found and fixed the problem!